### PR TITLE
JWS support

### PR DESCRIPTION
**Pull request by reinkrul**
This PR adds functionality to sign data using JWS with an ephemeral key pair + certificate, meaning it's only valid for a very short time, just long enough to span the moment of signing. The key pair isn't stored.

The advantage of this is that the CA doesn't need to care about maintaining a CRL for these certificates, since if the private key leaks, the validity period of the corresponding certificate is too short to be useful for exploits.

### CODE REVIEW

File: pkg/client.go [hunk 1]
```diff
@@ -49,6 +50,16 @@ type Client interface {
 	PublicKeyInJWK(legalEntity types.LegalEntity) (jwk.Key, error)
 	// SignJwtFor creates a signed JWT given a legalEntity and map of claims
 	SignJwtFor(claims map[string]interface{}, legalEntity types.LegalEntity) (string, error)
+	// JWSSignEphemeral signs payload according to the JWS spec with a temporary key and certificate which are generated just for this operation.
+	// In other words, the key and certificate are not stored and cannot be used for any other cryptographic operation.
+	// The certificate's validity is as short as possible, just spanning the instant of signing.
+	JWSSignEphemeral(payload []byte, ca types.LegalEntity, csr x509.CertificateRequest, signingTime time.Time) ([]byte, error)
```
Comment (woutslakhorst) [inline pkg/client.go:61]: Some documentation on the different arguments, at first glance it wasn't clear to me why a csr wass passed.

File: pkg/crypto.go [hunk 2]
```diff
@@ -138,22 +171,25 @@ func (client *Crypto) SignCertificate(entity types.LegalEntity, ca types.LegalEn
...
 	}
 	if profile.IsCA {
 		template.IsCA = true
 		template.MaxPathLen = profile.MaxPathLen
 		template.BasicConstraintsValid = true
 	}
 	var parentTemplate *x509.Certificate
-	if entity == ca {
-		// self-signed
+	if selfSigned {
```
Comment (woutslakhorst) [inline pkg/crypto.go:192]: When is selfSigned used?

Comment (reinkrul) [reply pkg/crypto.go:192]: When the vendor issues its vendor CA certificate.

File: pkg/client.go (hunk 1 above)
Comment (reinkrul) [reply pkg/client.go:61]: Added documentation for parameters

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**